### PR TITLE
Adding Samza SQL Examples

### PR DESCRIPTION
**Pull request by shenodaguirguis**

### CODE REVIEW

File: src/main/java/samza/examples/sql/samza-sql-filter/src/main/sql/samza.sql [hunk 1]
```diff
@@ -0,0 +1,11 @@
+-- Extract some fields from PageViewEvent and insert into a table.
```
Comment (atoomula) [inline src/main/java/samza/examples/sql/samza-sql-filter/src/main/sql/samza.sql]: Could you add this example instead ? Let's avoid espresso.

-- Filter Profile change-capture stream by 'Product Manager'
--  title and project basic profile data to a kafka topic.

INSERT INTO kafka.ProductManagerProfiles
     SELECT memberId, firstName, lastName, company
     FROM kafka.ProfileChanges
     WHERE standardize(title) = 'Product Manager'

File: src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql [hunk 2]
```diff
@@ -0,0 +1,5 @@
+insert into kafka.groupbyTopic
+  select GetPageKey(pv.requestHeader) as __key__, GetPageKey(pv.requestHeader) as pageKey, count(*) as Views
+  from kafka.tracking.`PageViewEvent` as pv
+  group by (GetPageKey(pv.requestHeader))
+-- Write additional SQL Code here
```
Comment (atoomula) [inline src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql]: Remove this line

File: src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql [hunk 3]
```diff
@@ -0,0 +1,5 @@
+insert into kafka.groupbyTopic
+  select GetPageKey(pv.requestHeader) as __key__, GetPageKey(pv.requestHeader) as pageKey, count(*) as Views
+  from kafka.tracking.`PageViewEvent` as pv
```
Comment (atoomula) [inline src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql]: Just kafka.PageViewEvent. tracking is linkedin specific cluster.

File: src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql [hunk 4]
```diff
@@ -0,0 +1,5 @@
+insert into kafka.groupbyTopic
```
Comment (atoomula) [inline src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql]: Could we add a comment as to what we are doing here ?

-- Emit Page view counts collected grouped by page key in the last
-- 5 minutes at 5 minute interval and send the result to a kafka topic.
-- Using GetSqlField UDF to extract page key from the requestHeader.

File: src/main/java/samza/examples/sql/samza-sql-samples-yarn/src/main/sql/samza.sql [hunk 5]
```diff
@@ -0,0 +1,3 @@
+insert into log.outputStream
```
Comment (atoomula) [inline src/main/java/samza/examples/sql/samza-sql-samples-yarn/src/main/sql/samza.sql]: I'm not sure if this example is required. We can delete it.

File: src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+  select GetPageKey(pv.requestHeader) as __key__, GetPageKey(pv.requestHeader) as pageKey, count(*) as Views
```
Comment (atoomula) [inline src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql]: Replace GetPageKey with GetSqlField udf.

File: src/main/java/samza/examples/sql/samza-sql-stream-table-join/src/main/sql/samza.sql [hunk 7]
```diff
@@ -0,0 +1,6 @@
+insert into log.outputStream
```
Comment (atoomula) [inline src/main/java/samza/examples/sql/samza-sql-stream-table-join/src/main/sql/samza.sql]: Could we replace this with:

-- Enrich PageViewEvent with member profile data

INSERT INTO kafka.tracking.EnrichedPageViewEvent
   SELECT *
   FROM kafka.PageViewEvent as pv
      JOIN kafka.ProfileChanges.`$table` as p
          ON pv.memberId = p.memberId

Comment (atoomula) [reply src/main/java/samza/examples/sql/samza-sql-stream-table-join/src/main/sql/samza.sql]: $table should have back ticks around it.

File: src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql (hunk 2 above)
Comment (shenodaguirguis) [reply src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql]: I added this line to emphasis that a sql file can contain multiple sql statements

File: src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql (hunk 4 above)
Comment (shenodaguirguis) [reply src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql]: done

File: src/main/java/samza/examples/sql/samza-sql-samples-yarn/src/main/sql/samza.sql (hunk 5 above)
Comment (shenodaguirguis) [reply src/main/java/samza/examples/sql/samza-sql-samples-yarn/src/main/sql/samza.sql]: removed

File: src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql (hunk 6 above)
Comment (shenodaguirguis) [reply src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql]: done

File: src/main/java/samza/examples/sql/samza-sql-stream-table-join/src/main/sql/samza.sql (hunk 7 above)
Comment (shenodaguirguis) [reply src/main/java/samza/examples/sql/samza-sql-stream-table-join/src/main/sql/samza.sql]: done

File: src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql (hunk 3 above)
Comment (shenodaguirguis) [reply src/main/java/samza/examples/sql/samza-sql-groupby/src/main/sql/samza.sql]: done

**Comment by shenodaguirguis** [comment]
@atoomula I think it is better to say we have complete basic operations but some are not fully stable yet as opposed to say we don't have complete set of operations.  So I suggest  keep these examples and add a comment that they are not fully stable in this release and that we are actively working on stabilizing them. I will still add the example you listed. What do you think?

**Comment by atoomula** [comment]
Sure.

Sent from my iPhone

On Dec 24, 2018, at 11:19 AM, Shenoda Guirguis <notifications@github.com<mailto:notifications@github.com>> wrote:


@atoomula<https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fatoomula&data=02%7C01%7Catoomula%40linkedin.com%7Cf3e03d1011484e026bed08d669d4b478%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C636812759614279721&sdata=Nyc5TuPOpf%2FWVMAzThR3Nf8gSzu5yYiMFEqZ%2F0px7GI%3D&reserved=0> I think it is better to say we have complete basic operations but some are not fully stable yet as opposed to say we don't have complete set of operations. So I suggest keep these examples and add a comment that they are not fully stable in this release and that we are actively working on stabilizing them. I will still add the example you listed. What do you think?

File: src/main/java/samza/examples/sql/samza-sql-filter/src/main/sql/samza.sql (hunk 1 above)
Comment (shenodaguirguis) [reply src/main/java/samza/examples/sql/samza-sql-filter/src/main/sql/samza.sql]: done

File: src/main/java/samza/examples/sql/samza-sql-casewhen/src/main/sql/samza.sql [hunk 8]
```diff
@@ -0,0 +1,12 @@
+-- For each profile in Kafka Profile change capture stream, identify whether the
+-- profile is a quality profile or not and insert the result into QualityProfile
+-- kafka topic. Please note the usage of GetSqlField to extract the company
```
Comment (atoomula) [inline src/main/java/samza/examples/sql/samza-sql-casewhen/src/main/sql/samza.sql]: GetSqlField udf

**Pull request closed without merging**